sign/dilithium/internal/common/poly.go

```diff
@@ -105,26 +105,42 @@ func (p *Poly) MakeHint(p0, p1 *Poly) (pop uint32) {
 // Computes corrections to the high bits of the polynomial q according
 // to the hints in h and sets p to the corrected high bits.  Returns p.
 func (p *Poly) UseHint(q, hint *Poly) *Poly {
+	var q0PlusQ Poly
+
+	// See useHint() and makeHint() for an explanation.  We reimplement it
+	// here so that we can call Poly.Decompose(), which might be way faster
+	// than calling decompose() in a loop (for instance when having AVX2.)
+
+	q.Decompose(&q0PlusQ, p)
+
 	for i := 0; i < N; i++ {
-		p[i] = useHint(q[i], hint[i])
+		if hint[i] == 0 {
+			continue
+		}
+		if q0PlusQ[i] > Q {
```

Comment:
This could be done in straight-line code.

Reply:
I don't understand your suggestion.  Could you elaborate?

Reply:
I was asking if you can replace the if-then-else by constant-time statements.

Reply:
For security, there is no requirement that this is constant-time.  The hint is basically a compression technique for the public signature.  Or do you think the constant-time version will perform better?

Reply:
It's ok as is.